utils/utils.go

```diff
@@ -190,3 +190,22 @@ func Trim(args []string) []string {
 	}
 	return out
 }
+
+// Join multiple lists of strings. This replaces appending multiple arrays
+// together before calling strings.Join().
+func Join(lists ...[]string) (joined string) {
+	first := true
+	for _, list := range lists {
+		for _, elem := range list {
```

Comment:
Should we be calling Strings.Join to join the content of each slice?
I guess we want to avoid re-allocating string buffers, but here we're constantly extending the string buffer for every element

Reply:
Good question...

I originally did it this way so that it would be easy to convert to use a Builder: https://golang.org/pkg/strings/#Builder
That would be possible to do if we eventually require Go 1.10.

But you're right, it would be faster for now to use strings.Join() on each element. It might be even better to count the total length at the start though and allocate the correct length for the entire final string - I'll try it and see. That's also probably not much worse than using a Builder.